core-scala/src/test/scala/com/baeldung/scala/forcomprehension/ForLoopTest.scala

```diff
@@ -0,0 +1,113 @@
+package com.baeldung.scala.forcomprehension
+
+import org.junit.Assert.assertEquals
+import org.junit.{Before, Test}
+
+class ForLoopImplementation extends ForLoop {
+  var sideEffectResult: Seq[Any] = Nil
+  override def sideEffectFunction (arg: Any): Unit = {
+    sideEffectResult = sideEffectResult :+ arg
+  }
+
+  def clean(): Unit = {
+    sideEffectResult = Nil;
+  }
+}
+
+class ForLoopTest {
```

Comment:
It seems that the *UnitTest naming convention applies for the Scala repo as well -- all the current test class files are named that way.